pkg/reconciler/route/route.go

```diff
@@ -258,18 +259,18 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 	// and may not have had all of the assumed defaults specified.  This won't result
 	// in this getting written back to the API Server, but lets downstream logic make
 	// assumptions about defaulting.
-	r.SetDefaults(ctx)
+	r.SetDefaults(v1beta1.WithUpgradeViaDefaulting(ctx))
 	r.Status.InitializeConditions()
 
-	// There are no conditions that would trigger this, but if they were we'd have a
```

Comment:
Should we just make reconciler read/write beta resources then?

Reply:
We don't expose v1beta1 yet, this is to ensure things are within the v1beta1 subset of v1alpha1 so that we can use schema-equivalence.